### PR TITLE
refactor bucket lifecycle apis  to ts

### PR DESCRIPTION
**Pull request by prakashsvmx**
refactor
`setBucketLifecycle` 
`getBucketLifecycle` 
`removeBucketLifecycle` 

apis to typescript.

### CODE REVIEW

File: src/internal/client.ts [hunk 1]
```diff
@@ -2290,4 +2291,52 @@ export class TypedClient {
...
+  async removeBucketLifecycle(bucketName: string): Promise<void> {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
+    }
+    const method = 'DELETE'
+    const query = 'lifecycle'
+    await this.makeRequestAsyncOmit({ method, bucketName, query }, '', [204])
+  }
+
+  async setBucketLifecycle(bucketName: string, lifeCycleConfig: LifecycleConfig | null): Promise<void> {
```
Comment (kaankabalak) [inline src/internal/client.ts:2320]: Would we need to add the empty string to the type of `lifeCycleConfig` as an empty string is also a possible argument for this parameter according to the [documentation](https://github.com/minio/minio-js/pull/1276/files#diff-b57590968a12cee85a37c1b91d8cc7092cd8b68e50b242c39a178121db82a797R646)?

```suggestion
  async setBucketLifecycle(bucketName: string, lifeCycleConfig: LifecycleConfig | null | ''): Promise<void> {
```

File: examples/set-bucket-lifecycle.mjs [hunk 2]
```diff
@@ -40,12 +40,8 @@ const lifecycleConfig = {
...
 }
 
-s3Client.setBucketLifecycle('bucketname', lifecycleConfig, function (err) {
-  if (err) {
-    return console.log(err)
-  }
-  console.log('Success')
-})
+const ilmExpiryDays = await s3Client.setBucketLifecycle('test-bucket', ilmPolicyExpDays)
+console.log(ilmExpiryDays)
```
Comment (kaankabalak) [inline examples/set-bucket-lifecycle.mjs:44]: The `console.log` statements in this example seem to be returning `undefined` for me. This is expected, right?

File: examples/set-bucket-lifecycle.mjs [hunk 3]
```diff
@@ -67,9 +63,5 @@ const lifecycleConfigWithExpirationDate = {
...
 }
 
-s3Client.setBucketLifecycle('bucketname', lifecycleConfigWithExpirationDate, function (err) {
-  if (err) {
-    return console.log(err)
-  }
-  console.log('Success')
-})
+const ilmDatePolicyRes = await s3Client.setBucketLifecycle('test-bucket', lifecycleConfigWithExpirationDate)
+console.log(ilmDatePolicyRes)
```
Comment (kaankabalak) [inline examples/set-bucket-lifecycle.mjs:67]: In the same vein as my previous comment, the `console.log` statements in this example seem to be returning `undefined` for me. This is expected, right?

File: examples/get-bucket-lifecycle.mjs [hunk 4]
```diff
@@ -25,9 +25,8 @@ const s3Client = new Minio.Client({
...
 })
 
-s3Client.removeBucketLifecycle('bucketname', function (err) {
-  if (err) {
-    return console.log(err)
-  }
-  console.log('Success')
-})
+try {
+  await s3Client.getBucketLifecycle('test-bucket')
```
Comment (kaankabalak) [inline examples/get-bucket-lifecycle.mjs:29]: Would it make sense to have a `console.log` statement here to display the bucket lifecycle configuration that was just retrieved?

```suggestion
  const bucketLifecycleConfiguration = await s3Client.getBucketLifecycle('test-bucket')
  console.log(bucketLifecycleConfiguration)
```

Comment (prakashsvmx) [reply examples/get-bucket-lifecycle.mjs]: Yes. Thank you. i have updated as per suggestions.

**Review by kaankabalak — APPROVED** [review]
LGTM with a very small note: Seems like the commit title exceeds the 72 character limit, would it be possible to edit that before merging?

**Pull request merged**